src/utils.js

```diff
@@ -4,3 +4,41 @@ module.exports.generateRandomKey = function (len) {
   len = len || 10;
   return Math.random().toString(36).substring(2).substring(0, len);
 };
+
+module.exports.deepGet = function(collection, path){
```

Comment:
even without the keyPath as string deepGet and deepSet or something like them are needed for dealing with the tree structure in a clean way, perhaps it can be eliminated, will look into it.